ui/shared/ad/CustomAdBanner.tsx

```diff
@@ -0,0 +1,61 @@
+import { Flex, chakra, Tooltip, Image } from '@chakra-ui/react';
+import { useQuery } from '@tanstack/react-query';
+import React, { useState, useEffect } from 'react';
+
+import type { AdCustomConfig } from 'types/client/ad';
+
+import config from 'configs/app';
+import type { ResourceError } from 'lib/api/resources';
+import useFetch from 'lib/hooks/useFetch';
+
+import useIsMobile from '../../../lib/hooks/useIsMobile';
+
+const CustomAdBanner = ({ className }: { className?: string }) => {
+  const isMobile = useIsMobile();
+
+  const feature = config.features.adsBanner;
+  const configUrl = (feature.isEnabled && feature.provider === 'custom') ? feature.configUrl : '';
+
+  const apiFetch = useFetch();
+  const { data: adConfig } = useQuery<unknown, ResourceError<unknown>, AdCustomConfig>(
+    [ 'custom-configs' ],
+    async() => apiFetch(configUrl),
+    {
+      enabled: feature.isEnabled && feature.provider === 'custom',
+      staleTime: Infinity,
+    });
+  const interval = adConfig?.interval || 60000;
```

Comment:
```js
import { MINUTE } from 'lib/consts';
const interval = adConfig?.interval || MINUTE;
```

just for the sake of code readability

Reply:
Yes, your reminder was great. I should learn more about some existing configurations in lib.